hotp/ta/hotp_ta.c

```diff
@@ -0,0 +1,234 @@
+/*
+ * Copyright (c) 2017, Linaro Limited
+ * All rights reserved.
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+#include <hotp_ta.h>
+#include <string.h>
+#include <tee_internal_api_extensions.h>
+#include <tee_internal_api.h>
+
+/*
+ * This macro checks whether the return value is not equal to TEE_SUCCESS, if it
+ * is not equal, then it will set the return value to 'res' and then goto
+ * 'label'.
+ */
+#define CHECK_EXIT(label, res, ret_val) \
```

Comment:
I have my doubts about these two macros. Personally I don't think it improves readability, and also when reading the code it's not so clear what's happening.

Then there's also the issue with `ret_val` being evaluated three times, doesn't seem proper since it's really a function call with side effects.

Reply:
I prefer to keep it, since I think it makes it easier to grasp the overall code since you will see all the GP magic in 15 lines, which otherwise would have spread out over ~30 lines instead. 

> @jenswi-linaro Then there's also the issue with ret_val being evaluated three times, doesn't seem proper since it's really a function call with side effects.

Ah! True, that must be changed.

Reply:
I'll remove the macro's so everyone is happy.